test-app/tests/helpers/utility-methods.js

```diff
@@ -20,139 +24,144 @@ export function fetchJSON({ url, params, method = 'GET' } = {}) {
   );
 }
 
-//// custom adapter options for the various models
-//// which are applied to the currently testing model's adapter ( JSONAPI, REST, ActiveModel, etc )
-//const adapterOptions = {
-//  employee: {
-//    buildURL(modelName, id, snapshot, requestType, query)  {
-//      const url = this._super(modelName, id, snapshot, requestType, query);
-//      const delimiter = (url.indexOf('?') !== -1) ? '&' : '?';
-//      return `${url}${delimiter}company_id=12345`;
-//    },
-//    urlForFindRecord(id, modelName, snapshot) {
-//      if (id === 'self') {
-//        return '/user';
-//      } else {
-//        return this._super(id, modelName, snapshot);
-//      }
-//    }
-//  }
-//};
-
-// custom serializer options for the various models
-// which are applied to the currently testing  model's serializer ( JSONAPI, REST, ActiveModel, etc )
-const serializerOptions = {
-  'entry-type': {
-    attrs: {
-      entries: { serialize: true },
-    },
-    // don't pluralize the payload key.
-    payloadKeyFromModelName(modelName) {
-      return modelName;
-    },
-  },
-  entry: {
-    // don't pluralize the payload key.
-    payloadKeyFromModelName(modelName) {
-      return modelName;
-    },
-  },
-  dog: {
-    primaryKey: 'dogNumber',
-    attrs: {
-      owner: { key: 'humanId' },
-    },
-  },
-  cat: {
-    primaryKey: 'catId',
-    attrs: {
-      name: { key: 'catName' },
-      friend: 'catFriend',
-    },
-  },
-  'comic-book': [
-    EmbeddedRecordsMixin,
-    {
-      attrs: {
-        includedVillains: { embedded: 'always' },
-        company: { embedded: 'always' },
-        characters: { embedded: 'always' },
-      },
-    },
-  ],
-};
-
-//function setupCustomAdapter(container, adapterType, options) {
-//  let store = container.lookup('service:store');
-//  let modelAdapter = container.lookup('adapter:' + adapterType);
-//  if (typeOf(options) === 'array') {
-//    modelAdapter.reopen.apply(modelAdapter, options);
-//  } else {
-//    modelAdapter.reopen(options);
-//  }
-//  modelAdapter.store = store;
-//  return modelAdapter;
-//}
-
-function setupCustomSerializer(container, serializerType, options) {
-  let store = container.lookup('service:store');
-  let modelSerializer = container.lookup('serializer:' + serializerType);
-  if (typeOf(options) === 'array') {
-    modelSerializer.reopen.apply(modelSerializer, options);
-  } else {
-    modelSerializer.reopen(options);
```

Comment:
use of reopen is removed in ember 5.x, so this is now all done as native classes with extends